panel/config.py

```diff
@@ -433,19 +433,24 @@ def __getattribute__(self, attr):
         """
         from .io.state import state
 
-        if attr == '_param__private':
-            return super().__getattribute__('_param__private')
+        if attr in ('_param__private', '__class__', 'npm_cdn', 'param'):
+            return super().__getattribute__(attr)
 
         # _param__private added in Param 2
         try:
             init = super().__getattribute__('_param__private').initialized
         except AttributeError:
             init = super().__getattribute__('initialized')
-        global_params = super().__getattribute__('_globals')
+
+        global _CONFIG_PARAMS
+        if not _CONFIG_PARAMS and init:
+            _CONFIG_PARAMS = set(super().__getattribute__('param'))
+
```

Comment:
I don't understant if I need to handle `attr.startswith('__')` as a special case. Or if I could just throws `params` way and refer to `_PARAMS`.